### ISSUE TITLE
Subshell Expansions Often Highlight as Strings

### ISSUE THREAD

**Issue by stevekuznetsov** [jfcherng/Sublime-ShellScriptImproved#29]
I'm not sure if this is intended or not, but for example:

``` sh
#!/bin/bash

function my_func() {
    echo "hello!"
}

output="$( my_func )"
output=$( cat <<<"$( my_func )" )
output=$( my_func "$( my_func )" )
output=$( cat <( my_func ) )
```

The only literal string that's present is `"hello!"` but the following is how this is highlighted:
![myfunc](https://cloud.githubusercontent.com/assets/7328370/19195505/eea62326-8c7f-11e6-8aab-73c03c411815.png)

**Comment by stevekuznetsov** [comment]
Quoting the expansions as in the following doesn't change the highlighting:

``` sh
#!/bin/bash

function my_func() {
    echo "hello!"
}

output="$( my_func )"
output="$( cat <<<"$( my_func )" )"
output="$( my_func "$( my_func )" )"
output="$( cat <( my_func ) )"
```

**Comment by jfcherng** [comment]
TL;DR
wbond (one of ST maintainers) and facelessuser (the author of BH) said they have a way to fix this issue. And I am just waiting for it.

## 

Yes. Currently we have some problems about this. The root cause is that I try to use `meta.reset.color` to get things back to their default color (mostly white for a dark theme). In recent ST release, `clear_scope` is introduced and it can indeed resolve this in a decent way. But, unfortunately, that way breaks the way BracketHighlighter(BH) works. BH is a top-25 plugin so the fix may not be acceptable. People in ST forum thought this issue could be resolved so I am waiting for an example of a solution.

## 

Ref: 
- [Problem with clear_scopes and BracketHighlighter](https://forum.sublimetext.com/t/problem-with-clear-scopes-and-brackethighlighter/23267)
- An [experimental branch](https://github.com/jfcherng/Sublime-ShellScriptImproved/tree/clear_scopes) with `clear_scope` used.

**Comment by stevekuznetsov** [comment]
@jfcherng understood

**Comment by jfcherng** [comment]
I have tweak the scope in `1.2.21` to do coloring better.
But this issue still remains in more complex codes (in that case, codes are written badly I guess).

![image](https://cloud.githubusercontent.com/assets/6594915/25057745/1678766e-21a5-11e7-8709-265c7e9cba3f.png)

Feel free to report here if this issue remains.

##

If you didn't use the bundled color scheme in this package, you would have to add the following settings into your color scheme.

```xml
<dict>
    <!-- force codes to be shown as the default color (white) -->
    <key>name</key>
    <string>Embedded</string>
    <key>scope</key>
    <string>meta.embedded</string>
    <key>settings</key>
    <dict>
        <key>foreground</key>
        <string>#FFFFFF</string>
    </dict>
</dict>
```

**Comment by stevekuznetsov** [comment]
Cool, thanks!
